debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractTransactionCachingLogMinerEventProcessor.java

```diff
@@ -0,0 +1,483 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.logminer.processor;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.connector.oracle.OracleConnection;
+import io.debezium.connector.oracle.OracleConnectorConfig;
+import io.debezium.connector.oracle.OracleDatabaseSchema;
+import io.debezium.connector.oracle.OracleOffsetContext;
+import io.debezium.connector.oracle.OraclePartition;
+import io.debezium.connector.oracle.Scn;
+import io.debezium.connector.oracle.logminer.LogMinerStreamingChangeEventSourceMetrics;
+import io.debezium.connector.oracle.logminer.events.LogMinerEvent;
+import io.debezium.connector.oracle.logminer.events.LogMinerEventRow;
+import io.debezium.pipeline.EventDispatcher;
+import io.debezium.pipeline.source.spi.ChangeEventSource.ChangeEventSourceContext;
+import io.debezium.relational.TableId;
+import io.debezium.util.Loggings;
+
+/**
+ * An implementation of {@link LogMinerEventProcessor}
+ * that uses Infinispan to persist the transaction cache across restarts on disk.
+ *
+ * @author Chris Cranford
+ */
+public abstract class AbstractTransactionCachingLogMinerEventProcessor<T extends Transaction> extends AbstractLogMinerEventProcessor<T> implements CacheProvider<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractTransactionCachingLogMinerEventProcessor.class);
+    private final OracleConnection jdbcConnection;
+    private final LogMinerStreamingChangeEventSourceMetrics metrics;
+    private final OraclePartition partition;
+    private final OracleOffsetContext offsetContext;
+    private final EventDispatcher<OraclePartition, TableId> dispatcher;
+
+    private InMemoryPendingTransactionsCache inMemoryPendingTransactionsCache = new InMemoryPendingTransactionsCache();
+
+    private static AbstractTransactionCachingLogMinerEventProcessor instance;
```

Comment:
This will get us in trouble in Kafka Connect where multiple connector instances will run.

Reply:
The instance only seems to be used by logCacheStats which has a comment about being useful for Debezium embedded mode.  Doesn't appear to be used by DBZ itself.  Should we just remove it?

Reply:
Oh, I didn't realize that was part of the current code base :thinking: 

I wouldn't change necessarily in this work then for EhCache, let me add a Jira and take care of that on the main branch because we should definitely fix this asap.